core/new-gui/src/app/dashboard/service/workflow-version/workflow-version.service.ts

```diff
@@ -52,9 +60,80 @@ export class WorkflowVersionService {
     this.setDisplayParticularVersion(true);
     // disable modifications because it is read only
     this.workflowActionService.disableWorkflowModification();
+    // highlight the different elements
+    this.highlightDifference(this.highlightedIDs);
+  }
+
+  public getPositionDifference(v1: Workflow, v2: Workflow) {
+    var p1 = v1.content.operatorPositions;
+    var p2 = v2.content.operatorPositions;
+    var difference = reduce(
+      p2,
+      function (result: string[], value, key) {
+        return p1[key as keyof typeof p1] == undefined || isEqual(value, p1[key as keyof typeof p1])
+          ? result
+          : result.concat(key);
+      },
+      []
+    );
+    return difference;
+  }
+
+  public highlightDifference(differences: any[]) {
+    for (var diff of differences) {
+      this.workflowActionService
+        .getJointGraphWrapper()
+        .getMainJointPaper()
+        ?.getModelById(diff)
+        .attr("rect.boundary/fill", "rgba(255,118,20,0.5)");
+    }
+  }
+
+  public getDifference(v1: Workflow, v2: Workflow) {
+    var typeToInclude = ["operators", "links", "groups", "commentBoxes", "breakpoints"];
+    var diff: any[] = [];
+    var c1 = v1.content;
+    var c2 = v2.content;
+    var differentTypes = reduce(
+      c2,
+      function (result: string[], value, key) {
+        return isEqual(value, c1[key as WorkflowContents]) ? result : result.concat(key);
+      },
+      []
+    );
+    for (var i = 0; i < differentTypes.length; i++) {
+      var type = differentTypes[i];
+      if (typeToInclude.includes(type)) {
+        let getIDList = function (element: Elements, index: number) {
+          var id = "";
+          if (type.substring(type.length - 2, type.length) === "es") {
+            id = type.substring(0, type.length - 2) + "ID";
+          } else {
+            id = type.substring(0, type.length - 1) + "ID";
+          }
+          return { id: element[id as keyof Elements], index: index };
+        };
+        var l1 = c1[type as keyof WorkflowContent] as Elements[];
+        var IDList1 = l1.map(getIDList);
+        var l2 = c2[type as keyof WorkflowContent] as Elements[];
+        var IDList2 = l2.map(getIDList);
+        var added = IDList2.filter(x => !IDList1.map(y => y.id).includes(x.id)).map(x => x.id);
```

Comment:
maybe you can iterate the list once by passing a function that satisfies both your conditions